src/components/QuestionPageComponents/UploadJsonButton.tsx

```diff
@@ -0,0 +1,128 @@
+import * as React from 'react';
+import UploadFileIcon from '@mui/icons-material/UploadFile';
+import CloudUploadIcon from '@mui/icons-material/CloudUpload';
+import { Box, Button, Modal, Stack, Typography } from '@mui/material';
+import { CopyBlock, dracula } from 'react-code-blocks';
+import axios from 'axios';
+
+const codeBlockProps = {
+    language: "json",
+    showLineNumbers: true,
+    wrapLongLines: true,
+    theme: dracula,
+    code: '{\n  "questions": [\n        {\n         "question_id": 1,\n            "question_title": "Add Binary",\n            "question_description": "Given two binary strings a and b, return their sum as a binary string.",\n            "question_categories": ["Bit Manipulation", "Algorithms"],\n            "question_complexity": "Easy",\n            "popularity": 10\n        },\n        {\n            "question_id": 2,\n            "question_title": "LRU Cache Design",\n            "question_description": "Design and implement an LRU (Least Recently Used) cache.",\n            "question_categories": ["Data Structures"],\n            "question_complexity": "Medium",\n            "popularity": 25\n        }\n    ]\n}',
+}
+
+const api = axios.create({
+    baseURL: "/* TODO */", // TODO
+    timeout: 5000,
+})
+
+const UploadJsonButton = () => {
+    const [open, setOpen] = React.useState(false);
+    const [file, setFile] = React.useState<File | null>(null);
+
+    const handleOpen = () => setOpen(true);
+    const handleClose = () => setOpen(false);
+
+    function handleAxiosError(error: any) {
+        console.error('An error occurred:', error.message);
+        return null;
+    }
+
+    const handleFileChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        if (e.target.files) {
+            setFile(e.target.files[0]);
+        }
+    };
+
+    const handleUpload = async () => {
+        const formData = new FormData();
+        if (file) {
+            formData.append('file', file);
+        }
+        console.log(formData);
+
+        try {
+            const response = await api.post("/", formData);
```

Comment:
Handling of axios should not be here. Should be done in the controller/question-service